### ISSUE TITLE
Using `\w` in Find dialog doesn't find non-English Unicode characters

### ISSUE THREAD

**Issue by ermshiperete** [microsoft/vscode#209811]
Does this issue occur when all extensions are disabled?: Yes

Version: 1.88.0
Commit: 5c3e652f63e798a5ac2f31ffd0d863669328dc4c
Date: 2024-04-03T13:25:57.039Z
Electron: 28.2.8
ElectronBuildId: 27744544
Chromium: 120.0.6099.291
Node.js: 18.18.2
V8: 12.0.267.19-electron.0
OS: Linux x64 6.5.0-26-generic

Steps to Reproduce:

1. Create a document containing the word `Käse`
2. Open Find dialog, check "Use regular expressions" and enter `\w` in the Find field

Expected behavior:
- All four characters are selected

Actual behavior:
- `ä` is not selected

It is possible that this behavior is by design. In that case the problem is the lack of documentation of the Regex syntax that vscode uses, at least [this page](https://code.visualstudio.com/docs/editor/codebasics?ref=arctype.com#_advanced-find-and-replace-options) doesn't have any details.

Searching on Microsoft's websites reveals [this](https://learn.microsoft.com/en-us/dotnet/standard/base-types/character-classes-in-regular-expressions#word-character-w) page which suggests that `\w` should include all lower- and uppercase letters.

I guess the problem is that vscode uses the ECMAScript flavor. So a workaround would be to use [`[\p{Ll}\p{Lu}\p{Lt}\p{Lo}\p{Nd}\p{Pc}\p{Lm}]`](https://learn.microsoft.com/en-us/dotnet/standard/base-types/regular-expression-options#ecmascript-matching-behavior) to match all upper- and lowercase letters instead of `\w`.

**Comment by IllusionMH** [comment]
Yes, that's the root cause and expected behavior - VS Code uses JS RegExp engine for in-editor find/replace.

However, if you only need letter (not `0-9_-`) in most cases you should be able to use simpler `\p{L}` or combine it with other classes
